File: backend/internal/models/production_plan.go
package models

type ProductionPlan struct {
	Id       string `json:"id" db:"id"`
	Date     string `json:"date" db:"date" binding:"required"`
	Product  string `json:"product" db:"product" binding:"required"`
	Money    string `json:"money" db:"money"`
	Quantity int    `json:"quantity" db:"quantity"`
	Type     string `json:"type" db:"type"`
}

File: frontend/src/types/productionPlan.ts
export interface IProductionPlan {
	id: string | null
	product: string | null
	quantity: number | null
	money: number | null
}

export type PlanType = 'annual' | 'shipment' | 'output'

export interface IProductionPlanDTO {
	id: string
	type: PlanType
	date: string
	product: string
	quantity?: number
	money: string
}

export interface IPlan {
	id: string
	date: string
	product: string
	count: number
	money: number
	planQuantity?: number
	planMoney: number
}
